jira/client.py

```diff
@@ -1722,6 +1722,48 @@ def create_customer_request(
         else:
             return Issue(self._options, self._session, raw=raw_issue_json)
 
+    def createmeta_issuetypes(
+        self,
+        projectIdOrKey: Union[str, int],
+    ) -> Dict[str, Any]:
+        """Get the issue types metadata for a given project, required to create issues.
+
+        This API was introduced in JIRA 8.4 as a replacement for the more general purpose API 'createmeta'.
+        For details see: https://confluence.atlassian.com/jiracore/createmeta-rest-endpoint-to-be-removed-975040986.html
+
+        Args:
+            projectIdOrKey (Union[str, int]): id or key of the project for which to get the metadata.
+        Returns:
+            Dict[str, Any]
+        """
+        if self._version < (8, 4, 0):
+            raise JIRAError(f"Unsupported JIRA version: {self._version}")
+
+        return self._get_json(f"issue/createmeta/{projectIdOrKey}/issuetypes")
+
+    def createmeta_fieldtypes(
+        self,
+        projectIdOrKey: Union[str, int],
+        issueTypeId: Union[str, int],
+    ) -> Dict[str, Any]:
+        """Get the field metadata for a given project and issue type, required to create issues.
+
+        This API was introduced in JIRA 8.4 as a replacement for the more general purpose API 'createmeta'.
+        For details see: https://confluence.atlassian.com/jiracore/createmeta-rest-endpoint-to-be-removed-975040986.html
+
+        Args:
+            projectIdOrKey (Union[str, int]): id or key of the project for which to get the metadata.
+            issueTypeId (Union[str, int]): id of the issue type for which to get the metadata.
+        Returns:
+            Dict[str, Any]
+        """
+        if self._version < (8, 4, 0):
+            raise JIRAError(f"Unsupported JIRA version: {self._version}")
```

Comment:
also add the info from below in the exception ... => please use createmeta_issuetypes and createmeta_fieldtypes